packages/filebrowser-extension/src/index.ts

```diff
@@ -541,6 +542,20 @@ function addCommands(
       return Promise.all(
         toArray(
           map(widget.selectedItems(), item => {
+            // check for nonstandard content type
+            const ft = widget.model.manager.registry.getFileTypeForModel(item);
+            if (ft && ft.contentType === 'dirlike') {
+              // cast to FileBrowser-like
+              const browserlike = (ft.browser || widget) as SideBarWidget & {
+                model: FileBrowserModel;
+              };
+              const localPath = browserlike.model.manager.services.contents.localPath(
+                item.path
+              );
+              browserlike.activateInSidebar();
```

Comment:
I think that we shouldn't really be making assumptions about the layout of the shell. The shell might not implement sidebars, and the filebrowser might not be located in a sidebar. What is wrong with `app.shell.activateById(browserlike.id)`?

Reply:
This is the same issue of trying to get the code used for opening via "open" in the context menu (implemented here) to match the code used for opening via double click (implemented in `filebrowser/src/listing.ts`).

If the layout is significantly different, `activateInSidebar` will simply silently fail. I could change it so that in case of no sidebar found, it will fall back to the standard `Widget.activate`. Does that seem like a good idea to you?

Reply:
Alternative to the above, I could simply assume that `browserlike` always has an `IShell` somewhere in its ancestors, and rearrange `browserlike.activateInSidebar` to `browserlike.activateInShell`. 

`activateInShell` would always result in a call to `app.shell.activateById(browserlike.id)` or the equivalent, so long as the above assumption is satisfied.